File: gms-api/src/main/java/org/gms/pojo/Account.java
package org.gms.pojo;

import com.mybatisflex.annotation.Column;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import lombok.Data;

import java.util.Date;

/**
 * 比如有多数据源
 * 驼峰不转下划线
 */
@Table(value = "accounts", dataSource = "mysql", camelToUnderline = false)
@Data
public class Account {
    @Id(keyType = KeyType.Auto)
    private Integer id;
    private String name;
    private String password;
    private String pin;
    private String pic;
    @Column(value = "loggedin")
    private boolean loggedIn;
    @Column(value = "lastlogin")
    private Date lastLogin;
    @Column(value = "createdat")
    private Date createDate;
    private Date birthday;
    private boolean banned;
    @Column(value = "banreason")
    private String banReason;
    private String macs;
    private Integer nxCredit;
    private Integer maplePoint;
    private Integer nxPrepaid;
    @Column(value = "characterslots")
    private Integer characterSlots;
    private Integer gender;
    @Column(value = "tempban")
    private Date tempBan;
    @Column(value = "greason")
    private Integer gReason;
    private boolean tos;
    @Column(value = "sitelogged")
    private String siteLogged;
    @Column(value = "webadmin")
    private Integer webAdmin;
    private String nick;
    private Integer mute;
    private String email;
    private String ip;
    @Column(value = "rewardpoints")
    private Integer rewardPoints;
    @Column(value = "votepoints")
    private Integer votePoints;
    @Column(value = "hwid")
    private String hwId;
    private Integer language;
}

File: gms-ui/src/api/account.ts
import axios from 'axios';
import { isValidString } from '@/utils/stringUtils';
import { PageState } from '@/store/page';

export interface RegisterForm {
  name?: string;
  password?: string;
  checkPassword?: string;
  birthday?: string;
  language?: number;
}

export interface GMUpdateForm {
  newPwd?: string;
  newPwdCheck?: string;
  pin?: string;
  pic?: string;
  birthday?: string;
  nxCredit?: number;
  maplePoint?: number;
  nxPrepaid?: number;
  characterslots?: number;
  gender?: number;
  webadmin?: boolean;
  nick?: string;
  mute?: boolean;
  email?: string;
  rewardpoints?: number;
  votepoints?: number;
  language?: number;
}

export function getAccountList(
  page: number,
  size: number,
  id?: number,
  name?: string,
  lastLoginStart?: string,
  lastLoginEnd?: string,
  createdAtStart?: string,
  createdAtEnd?: string
) {
  let url = `/account/v1?page=${page}&size=${size}`;
  if (id !== undefined && id > 0) url += `&id=${id}`;
  if (isValidString(name)) url += `&name=${name}`;
  if (isValidString(lastLoginStart)) url += `&lastLoginStart=${lastLoginStart}`;
  if (isValidString(lastLoginEnd)) url += `&lastLoginEnd=${lastLoginEnd}`;
  if (isValidString(createdAtStart)) url += `&createdAtStart=${createdAtStart}`;
  if (isValidString(createdAtEnd)) url += `&createdAtEnd=${createdAtEnd}`;
  return axios.get<PageState>(url);
}

export function addAccount(data: RegisterForm) {
  return axios.post('/account/v1', data);
}

export function updateAccountByGM(id: number, data: GMUpdateForm) {
  return axios.put(`/account/v1/${id}`, data);
}

export function deleteAccount(id: number) {
  return axios.delete(`/account/v1/${id}`);
}

export function banAccount(id: number, reason?: string) {
  return axios.put(`/account/v1/${id}/ban`, { reason });
}

export function unbanAccount(id: number) {
  return axios.put(`/account/v1/${id}/unban`);
}

export function resetLoggedIn(id: number) {
  return axios.put(`/account/v1/${id}/reset/logged`);
}
